### PR TITLE
[Swap] Limit decimal for user input

### PR DESCRIPTION
**Pull request by veado**
- [x] Limit decimal for user inputs in Swap 
- [x] Fix `convertBaseAmountDecimal` to always round down  
- [x] Add `baseAmountForThorchain` incl. test

Part of #1139

### CODE REVIEW

File: src/renderer/helpers/assetHelper.test.ts [hunk 1]
```diff
@@ -192,16 +193,30 @@ describe('helpers/assetHelper', () => {
...
       const amount = baseAmount('123456', 6)
       const result = convertBaseAmountDecimal(amount, 6)
       expect(eqBaseAmount.equals(result, amount)).toBeTruthy()
     })
   })
+
+  describe('baseAmountForThorchain', () => {
+    it('converts 1e12 to 1e8', () => {
+      const result = baseAmountForThorchain(baseAmount('123456789012', 12))
+      console.log('result:', result.amount().toString())
```
Comment (sarawut11) [inline src/renderer/helpers/assetHelper.test.ts:213]: clean up console.log

File: src/renderer/helpers/assetHelper.test.ts [hunk 2]
```diff
@@ -192,16 +193,30 @@ describe('helpers/assetHelper', () => {
...
+      expect(eqBaseAmount.equals(result, baseAmount('12345678', 8))).toBeTruthy()
+    })
+    it('keeps 1e6 unchanged', () => {
+      const result = baseAmountForThorchain(baseAmount('123456', 6))
+      console.log('result:', result.amount().toString())
```
Comment (sarawut11) [inline src/renderer/helpers/assetHelper.test.ts:218]: clean up console.log

File: src/renderer/helpers/assetHelper.test.ts [hunk 3]
```diff
@@ -192,16 +193,30 @@ describe('helpers/assetHelper', () => {
     })
     it('converts 1e8 decimal to 1e6 ', () => {
       const result = convertBaseAmountDecimal(baseAmount('12345678', 8), 6)
-      expect(eqBaseAmount.equals(result, baseAmount('123457', 6))).toBeTruthy()
+      expect(eqBaseAmount.equals(result, baseAmount('123456', 6))).toBeTruthy()
     })
     it('converts 1e18 decimal to 1e8', () => {
       const result = convertBaseAmountDecimal(baseAmount('123456789012345678', 18), 8)
-      expect(eqBaseAmount.equals(result, baseAmount('12345679', 8))).toBeTruthy()
+      console.log('result:', result.amount().toString())
```
Comment (sarawut11) [inline src/renderer/helpers/assetHelper.test.ts:200]: clean up console.log

**Review by sarawut11 — APPROVED** [review]
LGTM, just small comments

**Review by junkai121 — DISMISSED** [review]
LGTM!

**Pull request merged**